src/app/logic/player.js

```diff
@@ -0,0 +1,44 @@
+export class Player {
+  constructor() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+    this.endGameData = {};
+  }
+
+  registerAnswer(correctAnswer, playerAnswer, isanswerCorrect) {
+    if (this.playerAnswer === false) {
+      throw new Error(
+        'error with playerAnswer, playerAnswer is null or undefined',
+      );
+    }
+    if (isanswerCorrect) {
+      this.correctAnswersScore += 1;
+    }
+    this.detailQuestionData.push({
+      playerAnswer,
+      isAnswerCorrect: isanswerCorrect,
+      answer: correctAnswer,
+    });
+    this.amountOfAnsweredQuestions += 1;
+  }
+
+  clearGameData() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+  }
+
+  createEndGameData() {
+    this.endGameData.correctAnswersScore = this.correctAnswersScore;
+    this.endGameData.amountOfAnsweredQuestions = this.amountOfAnsweredQuestions;
+    this.endGameData.detailQuestionData = this.detailQuestionData;
```

Comment:
I am wondering, should all these be kept in the Player object? Perhaps somewhere in the game? Player need to know the score , but does it have to know detailed questions?

Reply:
changed! will be in new commit